homeassistant/components/hassmpris/config_flow.py

```diff
@@ -0,0 +1,381 @@
+            if existing_entry := await self.async_set_unique_id(zeroconf_uid):
+                _LOGGER.debug("Existing entry, unique ID: %s", existing_entry.unique_id)
+                assert existing_entry.unique_id is not None  # for type checker
+                await persist_cert_data(existing_entry.unique_id)
+                self.hass.config_entries.async_update_entry(existing_entry, data=data)
+                await self.hass.config_entries.async_reload(existing_entry.entry_id)
+                return self.async_abort(reason="reauth_successful")
+
+        # Update existing entry that has the same host / ports.
+        if existing_entry := await self.async_set_unique_id(
+            self._get_unique_id_by_connection_data()
+        ):
+            _LOGGER.debug("Existing entry, generated ID: %s", existing_entry.unique_id)
+            assert existing_entry.unique_id is not None  # for type checker
+            await persist_cert_data(existing_entry.unique_id)
+            self.hass.config_entries.async_update_entry(existing_entry, data=data)
+            await self.hass.config_entries.async_reload(existing_entry.entry_id)
+            return self.async_abort(reason="reauth_successful")
+
+        # OK, no entries found with those two identifiers.
+        # Let's set a unique ID.
+        await self.async_set_unique_id(self._get_any_unique_id())
+        _LOGGER.debug("New entry")
+        assert self._title, "Impossible: the title is %r" % self._title
+        await persist_cert_data(self._get_any_unique_id())
+        return self.async_create_entry(
+            title=self._title,
+            data=data,
+        )
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        if not user_input:
+            return self.async_show_form(
+                step_id=STEP_USER,
+                data_schema=STEP_USER_DATA_SCHEMA,
+                errors={},
+            )
+
+        self._set_data(
+            f"MPRIS on {user_input[CONF_HOST]}",
+            user_input[CONF_HOST],
+            user_input[CONF_CAKES_PORT],
+            user_input[CONF_MPRIS_PORT],
+        )
+
+        # Do not proceed if the device is already configured by hand.
+        await self.async_set_unique_id(self._get_unique_id_by_connection_data())
+        self._abort_if_unique_id_configured()
+
+        return await self.async_step_pairing()
+
+    async def async_step_zeroconf(
+        self, discovery_info: zeroconf.ZeroconfServiceInfo
+    ) -> FlowResult:
+        """Handle zeroconf discovery."""
+        try:
+            assert discovery_info.host
+            assert discovery_info.port
+            assert discovery_info.hostname
+            assert discovery_info.name
+            assert discovery_info.properties
+            assert CONF_CAKES_PORT in discovery_info.properties
+        except AssertionError:
```

Comment:
This is not a good way of using assert, assert is meant for debugging and testing.
Do normal Python code instead:
```py
if not discovery_info.host or not .. :
    _LOGGER.debug("Ignoring invalid zeroconf announcement: %s", discovery_info)
    return self.async_abort(reason=REASON_INVALID_ZEROCONF)
```

Reply:
Thanks.